clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -38,12 +38,14 @@ bool CheckedRegionAdder::VisitCompoundStmt(CompoundStmt *S) {
       if(isParentChecked(DTN) && !isFunctionBody(S)) {
         auto Loc = S->getBeginLoc();
         Writer.InsertTextBefore(Loc, "_Unchecked ");
+        TouchedFiles.insert(getFileID(Loc, *Context));
```

Comment:
I am noticing that you are pairing a call to `Writer.InsertText...` with a call to `TouchedFiles.insert(...`. Would it be possible to make a wrapper for `Writer` that did this internally? Just pass it `Context` too and it can make the call to `TouchedFiles.insert` internally. Might be more reliable against future regressions.